xinference/_compat.py

```diff
@@ -70,6 +75,45 @@
 )
 
 
+class JSONSchema(TypedDict, total=False):
+    name: Required[str]
+    """The name of the response format.
+
+    Must be a-z, A-Z, 0-9, or contain underscores and dashes, with a maximum length
+    of 64.
+    """
+
+    description: str
+    """
+    A description of what the response format is for, used by the model to determine
+    how to respond in the format.
+    """
+
+    schema_: Optional[Dict[str, object]] = Field(alias="schema", default=None)  # type: ignore
+    """The schema for the response format, described as a JSON Schema object."""
+
+    strict: Optional[bool]
+    """Whether to enable strict schema adherence when generating the output.
+
+    If set to true, the model will always follow the exact schema defined in the
+    `schema` field. Only a subset of JSON Schema is supported when `strict` is
+    `true`. To learn more, read the
+    [Structured Outputs guide](https://platform.openai.com/docs/guides/structured-outputs).
+    """
+
+
+class ResponseFormatJSONSchema(TypedDict, total=False):
+    json_schema: Required[JSONSchema]
+
+    type: Required[Literal["json_schema"]]
+    """The type of response format being defined: `json_schema`"""
+
+
+ResponseFormat: TypeAlias = Union[
```

Comment:
Shall us use pydantic model instead of TypeDice, like using `create_model_from_typeddict`?

Reply:
This part of code  was copied from openai's code, but it breaks python3.9's support. I think creating a pydantic model is ok.

Reply:
Yeah, I see the other part of this file used pydantic, this should be more strict on check.